caas/kubernetes/provider/k8s.go

```diff
@@ -458,11 +499,22 @@ func (k *kubernetesClient) EnsureService(
 		}
 	}()
 
-	unitSpec, err := makeUnitSpec(params.PodSpec)
+	unitSpec, err :=
+		makeUnitSpec(appName, params.PodSpec)
 	if err != nil {
 		return errors.Annotatef(err, "parsing unit spec for %s", appName)
 	}
 
+	for _, c := range params.PodSpec.Containers {
+		if c.ImageDetails.Password != "" {
+			imageSecretName := appSecretName(appName, c.Name)
+			if err := k.EnsureSecret(imageSecretName, appName, &c.ImageDetails); err != nil {
+				return errors.Annotatef(err, "creating secrets for container: %s", c.Name)
+			}
+		}
+		cleanups = append(cleanups, func() { k.deleteSecret(appName, c.Name) })
```

Comment:
and doing the above will fix this bug - we only need cleanup if secret is created